examples/todolist/tests/bindings/test_todolist.py

```diff
@@ -23,3 +23,28 @@
 entry2 = TodoEntry("Test Ünicode hàndling in an entry can't believe I didn't test this at first 🤣")
 todo.add_entry(entry2)
 assert(todo.get_last_entry().text == "Test Ünicode hàndling in an entry can't believe I didn't test this at first 🤣")
+
+todo2 = TodoList()
+assert(todo != todo2)
+assert(todo is not todo2)
+
+assert(get_default_list() is None)
+
+set_default_list(todo)
+assert(todo.get_items() == get_default_list().get_items())
+# In the future, we'll maintain object identity over the FFI.
+#assert(get_default_list() is todo)
+#assert(get_default_list() is not todo2)
```

Comment:
Currently, every object instance returned over the FFI is a fresh foreign-language object. Mark's [prototype PR](https://github.com/mozilla/uniffi-rs/pull/420) implemented a reverse-identity-map for the Python backend using weak references and I think we could do something similar for the other languages, but on balance I think it will be better for us to try out the basic functionality without blocking on that extra complexity. Happy to work on it in a followup PR, and I'm open to the suggestion that we shouldn't land without it because this "every object is a new one" behaviour could be confusing for consumers.

(There's also the question of how a reverse-identity-map would interact with the Kotlin `destroy()`/`.use` pattern, which I really don't have my head around just yet.)

Reply:
SGTM - maybe even just take that "promise" and commented code out entirely? It sounds like a commitment that I don't think we need to make. Alternatively, water it down "we might..." and link to an issue?

Reply:
You're right, we should just delete the promise; if it's important to us we'll actually do it, and if it isn't important we don't need to make any promises about it.